sale_order_secondary_unit_price/views/sale_order_views.xml

```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="view_order_form" model="ir.ui.view">
+        <field name="name">sale.order.secondary.unit.price</field>
+        <field name="model">sale.order</field>
+        <field name="inherit_id" ref="sale.view_order_form"/>
+        <field name="groups_id" eval="[(4, ref('uom.group_uom'))]"/>
+        <field name="arch" type="xml">
+            <xpath expr="//field[@name='order_line']/form//field[@name='price_subtotal']" position="after">
+                <field name="secondary_unit_price" attrs="{'readonly': [('parent.state', 'in', ('done', 'cancel'))]}"/>
+            </xpath>
+            <xpath expr="//field[@name='order_line']/tree//field[@name='product_uom_qty']" position="before">
+                <field name="secondary_unit_price" attrs="{'readonly': [('parent.state', 'in', ('done', 'cancel'))]}"/>
+            </xpath>
```

Comment:
@nakometal An example of making `price_unit` read-only when `secondary_unit_price` is set.

```.xml
            <xpath expr="//field[@name='order_line']/tree//field[@name='price_unit']" position="attributes">
                <attribute name="attrs">{'readonly': [('secondary_unit_price', '!=', 0.0)]}</attribute>
            </xpath>
```